watertap/property_models/NDMA_prop_pack.py

```diff
@@ -532,13 +538,15 @@ def _mole_frac_phase_comp(self):
             doc="Mole fraction",
         )
 
-        def rule_mole_frac_phase_comp(b, j):
-            return b.mole_frac_phase_comp["Liq", j] == b.flow_mol_phase_comp[
-                "Liq", j
-            ] / sum(b.flow_mol_phase_comp["Liq", j] for j in b.params.component_list)
+        def rule_mole_frac_phase_comp(b, p, j):
+            return b.mole_frac_phase_comp[p, j] == b.flow_mol_phase_comp[p, j] / sum(
+                b.flow_mol_phase_comp[p, j] for j in b.params.component_list
+            )
 
         self.eq_mole_frac_phase_comp = Constraint(
-            self.params.component_list, rule=rule_mole_frac_phase_comp
+            self.params.phase_list,
```

Comment:
This is more best practice than necessary but it is generally better to use `phase_component_set` in these cases - it doesn't really affect this property package but in more complex property packages this will ensure that non invalid phase-component pairs end up in the indexing set.

There are a lot of places where this could be applied.

Reply:
So basically this would require structuring all the property packages like cryst_prop_pack? I think this would require a decent bit of effort, so maybe this should be a separate PR? Also, I think it twould requiring hard-coding some phases and components as constraints (e.g. ["Liq"], ["NaCl"]), which is frequently done in the cryst_prop_pack. I think Tim had said we wanted to try and move away from that notation?